File: scripts/boards/TraditionalBoard.js
import CellState from "../CellState.js";
import Board from "./Board.js";

export default class TraditionalBoard extends Board {
    createBoard(rows, cols) {
        let matrix = Array(rows).fill().map(() => Array(cols).fill(CellState.PIECE1));
        let i = this.getRandomInt(0, rows);
        let j = this.getRandomInt(0, cols);
        matrix[i][j] = CellState.EMPTY;
        return matrix;
    }
}

File: src/main/java/model/boards/TraditionalBoard.java
package model.boards;

import model.CellState;

public class TraditionalBoard extends Board {
    public CellState[][] createBoard(int rows, int cols) {
        CellState[][] matrix = new CellState[rows][cols];
        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[i].length; j++) {
                matrix[i][j] = CellState.PIECE1;
            }
        }
        int i = this.getRandomInt(0, rows);
        int j = this.getRandomInt(0, cols);
        matrix[i][j] = CellState.EMPTY;
        return matrix;
    }
}
